src/Controller/WorkQueueController.php

```diff
@@ -50,9 +51,38 @@ public function __construct(
 
     /**
      * @Route("/", name="workqueue_index")
+     * @Route("/customized-view/{viewId}", name="workqueue_customized_view")
      */
-    public function index(Request $request)
+    public function index(): Response
     {
+        $workQueueView = $this->em->getRepository(WorkqueueView::class)->findOneBy([
+            'user' => $this->getUserEntity(),
+            'defaultView' => 1
+        ]);
+
+        if ($workQueueView) {
+            $redirectUrl = $this->generateUrl('workqueue_customized_view', [
+                    'viewId' => $workQueueView->getId()]) . '?' . $workQueueView->getFiltersQueryParams();
```

Comment:
Instead of using `http_build_query` inside the `getFiltersQueryParams` method, you could just merge the parameter array with the `viewId` parameter. Any parameter not part of the route pattern will automatically be added on as query parameters.